### ISSUE TITLE
Github Packages

### ISSUE THREAD

**Issue by disambiguationuk** [GlitchEnzo/NuGetForUnity#382]
## Description

Hello, I'm trying to use this with a NuGet feed on Github Packages, when I update the nuget.config to one that works on a another .NET project I get a 404 error:

Unable to retrieve package list from https://nuget.pkg.github.com/[Redacted]/index.jsonSearch()?$filter=IsLatestVersion&$orderby=DownloadCount desc&$skip=0&$top=15&searchTerm=''&targetFramework=''&includePrerelease=false
System.Net.WebException: The remote server returned an error: (404) Not Found.
  at System.Net.HttpWebRequest.EndGetResponse (System.IAsyncResult asyncResult) [0x00058] in <ef151b6abb5d474cb2c1cb8906a8b5a4>:0 
  at System.Net.HttpWebRequest.GetResponse () [0x0000e] in <ef151b6abb5d474cb2c1cb8906a8b5a4>:0 
  at NugetForUnity.NugetHelper.RequestUrl (System.String url, System.String userName, System.String password, System.Nullable`1[T] timeOut) [0x000b9] in <ebb7710be32040ba9a0cb144ce2a8967>:0 
  at NugetForUnity.NugetPackageSource.GetPackagesFromUrl (System.String url, System.String username, System.String password) [0x0006c] in <ebb7710be32040ba9a0cb144ce2a8967>:0 
  at NugetForUnity.NugetPackageSource.Search (System.String searchTerm, System.Boolean includeAllVersions, System.Boolean includePrerelease, System.Int32 numberToGet, System.Int32 numberToSkip) [0x000ce] in <ebb7710be32040ba9a0cb144ce2a8967>:0 
UnityEngine.Debug:LogErrorFormat (string,object[])
NugetForUnity.NugetPackageSource:Search (string,bool,bool,int,int)
NugetForUnity.NugetHelper:Search (string,bool,bool,int,int)
NugetForUnity.NugetWindow:UpdateOnlinePackages ()
NugetForUnity.NugetWindow:Refresh (bool)
NugetForUnity.NugetWindow:DrawOnlineHeader ()
NugetForUnity.NugetWindow:DrawOnline ()
NugetForUnity.NugetWindow:OnGUI ()
UnityEngine.GUIUtility:ProcessEvent (int,intptr)

Same issue with index.json/Search()

Going straight to index.json it says   "version": "3.0.0-beta.1" at the top. Are Github Packages exclusively v3 and therefore unable to work with this?

**Comment by jwittner** [comment]
I'm not sure about Github packages, but NugetForUnity only supports NuGet V2 API at them moment. The index.json endpoints are usually V3 API endpoints, and you'd need to find out what the V2 endpoints would be for Github Packages.

**Comment by disambiguationuk** [comment]
It's normally the api/v2/ endpoint that you generate in your NuGet.config for nuget.org right? I'll see if I can do some more digging but Github Packages doesn't seem to be well documented. I'll post what I find regardless.

**Comment by jwittner** [comment]
That's right, for nuget.org it's the `api/v2/` endpoint.

**Comment by BenMakesGames** [comment]
I'm getting the same (or a similar?) error, when trying to install Autofac:

```
Unable to retrieve package list from http://www.nuget.org/api/v2/FindPackagesById()?id='System.Threading.Tasks.Extensions'&$orderby=Version asc
System.Net.WebException: The remote server returned an error: (400) Bad Request.
```

that looks like it's v2. when visiting the URL myself, the response includes this message:

`The combination of parameters provided to this OData endpoint is no longer supported. Please refer to the following URL for more information about this deprecation: https://aka.ms/nuget/odata-deprecation`

**Comment by jwittner** [comment]
Changes have been published as [Release 3.0.2](https://github.com/GlitchEnzo/NuGetForUnity/releases/tag/v3.0.2).

**Comment by jwittner** [comment]
Closed this as the original uri was using unsupported V3 routes.
